tremor-script/src/ast/query/raw.rs

```diff
@@ -181,6 +239,274 @@ impl<'script> Upable<'script> for OperatorDeclRaw<'script> {
+        &self,
+        decl_params: &Option<HashMap<String, Value<'script>>>,
+        stmt_params: Option<HashMap<String, Value<'script>>>,
+        helper: &Helper<'script, 'registry>,
+    ) -> Result<HashMap<String, Value<'script>>> {
+        match &decl_params {
+            // No params in the declaration
+            None => Ok(HashMap::new()),
+            Some(decl_params) => {
+                match stmt_params {
+                    // No params in the creation
+                    None => Ok(decl_params.clone()),
+                    // Merge elided params from the declaration.
+                    Some(mut stmt_params) => {
+                        for (param, value) in decl_params {
+                            if !stmt_params.contains_key(param) {
+                                stmt_params.insert(param.clone(), value.clone());
+                            }
+                        }
+                        // Make sure no new params were introduced in creation
+                        if stmt_params.len() == decl_params.len() {
+                            Ok(stmt_params)
+                        } else {
+                            error_generic(self, self, &"Unknown parameter", &helper.meta)
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn inline_params<'registry>(
+        params: Params<'script>,
+        sq_args: &HashMap<String, Value<'script>>,
+        helper: &mut Helper<'script, 'registry>,
+    ) -> Result<HashMap<String, Value<'script>>> {
+        match params {
+            // ALLOW: Only Raw params can be inlined.
+            Params::Processed(_) => unreachable!("Can't inline processed params."),
+            Params::Raw(params) => params
+                .into_iter()
+                .map(|(name, value)| {
+                    let value = value.up(helper)?;
+                    if let ImutExprInt::Path(Path::Reserved(ReservedPath::Args {
+                        segments, ..
+                    })) = &value
+                    {
+                        if let Segment::Id { key, .. } = &segments[0] {
+                            let arg = key.key();
+                            if let Some(arg_val) = sq_args.get(arg) {
+                                return Ok((name.to_string(), arg_val.clone()));
+                            }
+                        }
+                    }
+                    Ok((name.to_string(), value.try_into_value(helper)?))
+                })
+                .collect(),
+        }
+    }
+
+    fn inline_maybe_params<'registry>(
+        params: MaybeParams<'script>,
+        sq_args: &HashMap<String, Value<'script>>,
+        helper: &mut Helper<'script, 'registry>,
+    ) -> Result<Option<HashMap<String, Value<'script>>>> {
+        match params {
+            // ALLOW: Only Raw params can be inlined.
+            MaybeParams::Processed(_) => unreachable!("Can't inline processed params."),
```

Comment:
see above, let's expand why this isn't reachable and rather use an error then a crashing construct